main/options/options.cc

```diff
@@ -1204,23 +1204,21 @@ void readOptions(Options &opts,
             }
         }
 
-        if (raw.count("packager-layers") && !opts.stripePackages) {
-            // Default values are available at raw[...], but don't show up in raw.count(...),
-            // so we can use this as a way to check if the user passed in the flag
-            logger->error("--packager-layers can only be specified in --stripe-packages mode");
-            throw EarlyReturnWithCode(1);
-        }
-
-        if (opts.stripePackages) {
-            // TODO(neil): This regex was picked on a whim, so open to changing to be more or less restrictive based on
-            // feedback/usecases.
-            std::regex layerValid("[a-zA-Z0-9]+");
-            for (const string &layer : raw["packager-layers"].as<vector<string>>()) {
-                if (!std::regex_match(layer, layerValid)) {
-                    logger->error("--packager-layers must contain items that are alphanumeric.");
-                    throw EarlyReturnWithCode(1);
+        if (raw.count("packager-layers")) {
+            if (opts.stripePackages) {
+                // TODO(neil): This regex was picked on a whim, so open to changing to be more or less restrictive based
+                // on feedback/usecases.
+                std::regex layerValid("[a-zA-Z0-9]+");
+                for (const string &layer : raw["packager-layers"].as<vector<string>>()) {
+                    if (!std::regex_match(layer, layerValid)) {
+                        logger->error("--packager-layers must contain items that are alphanumeric.");
+                        throw EarlyReturnWithCode(1);
+                    }
+                    opts.packagerLayers.emplace_back(layer);
                 }
-                opts.packagerLayers.emplace_back(layer);
+            } else {
+                logger->error("--packager-layers can only be specified in --stripe-packages mode");
+                throw EarlyReturnWithCode(1);
             }
```

Comment:
Is this nested for stylistic reasons, or is there something about the implicit_value change that meant that the `raw.count` calls don't work the same way they use to, making this structure semantically different from the old, early-return form?

Reply:
Previously, it was possible for `raw.count("packager-layers")` to be 0, but `raw["packager-layers"]` still have values. Now, if `raw.count(...)` is 0, then `raw[...]` will also have no values, so we need to check the count first before looping through (`raw[...]` raises an exception if the given flag doesn't have a value).